src/slave_setup.c

```diff
@@ -0,0 +1,137 @@
+/***
+* FILENAME : slave_setup.c
+*
+* DESCRIPTION :
+*       Setting up STM32F as a Slave device. 
+*
+* PUBLIC FUNCTIONS :
+*       void     led_setup_i2c  ( void )
+*       void     led_on_i2c     ( char )
+*       void     led_off_i2c    ( char )
+*
+* AUTHOR : Brittney and Freddy
+***/
+
+#include <drivers/i2c.h>
+#include <stdio.h>
+#include <drivers/gpio.h>
+
+//LED CONFIG
+#define LED0_NODE DT_ALIAS(led0)
+#define LED0    DT_GPIO_LABEL(LED0_NODE, gpios)
+#define PIN0    DT_GPIO_PIN(LED0_NODE, gpios)
+#define FLAGS0    DT_GPIO_FLAGS(LED0_NODE, gpios)
+
+#define LED1_NODE DT_ALIAS(led1)
+#define LED1    DT_GPIO_LABEL(LED1_NODE, gpios)
+#define PIN1    DT_GPIO_PIN(LED1_NODE, gpios)
+#define FLAGS1    DT_GPIO_FLAGS(LED1_NODE, gpios)
+
+#define LED2_NODE DT_ALIAS(led2)
+#define LED2    DT_GPIO_LABEL(LED2_NODE, gpios)
+#define PIN2    DT_GPIO_PIN(LED2_NODE, gpios)
+#define FLAGS2    DT_GPIO_FLAGS(LED2_NODE, gpios)
+
+#define LED3_NODE DT_ALIAS(led3)
+#define LED3    DT_GPIO_LABEL(LED3_NODE, gpios)
+#define PIN3    DT_GPIO_PIN(LED3_NODE, gpios)
+#define FLAGS3    DT_GPIO_FLAGS(LED3_NODE, gpios)
+
+const struct device *sled_dev;
+const struct device *sled_dev1;
+const struct device *sled_dev2;
+const struct device *sled_dev3;
+
+#define I2C_LABEL DT_PROP(DT_NODELABEL(i2c1), label)
+
+void led_setup_i2c()
+{
+    sled_dev = device_get_binding(LED0);
+    sled_dev1 = device_get_binding(LED1);
+    sled_dev2 = device_get_binding(LED2);
+    sled_dev3 = device_get_binding(LED3);
+    gpio_pin_configure(sled_dev, PIN0, GPIO_OUTPUT_INACTIVE | FLAGS0);
+    gpio_pin_configure(sled_dev1, PIN1, GPIO_OUTPUT_INACTIVE | FLAGS1);
+    gpio_pin_configure(sled_dev2, PIN2, GPIO_OUTPUT_INACTIVE | FLAGS2);
+    gpio_pin_configure(sled_dev3, PIN3, GPIO_OUTPUT_INACTIVE | FLAGS3);
+
+}
+
+void led_on_i2c(char color)
+{
+    if(color == 'r')
+    {
+        gpio_pin_set(sled_dev, PIN0, (int)true); //red
+    }
+    else if(color == 'o')
+    {
+        gpio_pin_set(sled_dev1, PIN1, (int)true); //orange
+    }
+    else if(color == 'g')
+    {
+        gpio_pin_set(sled_dev2, PIN2, (int)true); //green
+    }
+    else if(color == 'b')
+    {
+        gpio_pin_set(sled_dev3, PIN3, (int)true); //blue
+    }
+    
+}
+
+void led_off_i2c(char color)
```

Comment:
Could we combine led_off_i2c with led_on_i2c, and pass in a bool for on or off? I don't see if led_off_i2c is ever used, so I think it'd be ok to combine them.

Reply:
I'm planning on leaving this it matches the can-receive structure for consistency